core/field_textinput.js

```diff
@@ -292,34 +292,33 @@ Blockly.FieldTextInput.prototype.widgetCreate_ = function() {
 };
 
 /**
- * Close the editor, save the results, and dispose any events bound to the
- * text input's editor.
+ * Closes the editor, saves the results, and disposes of any events or
+ * dom-references belonging to the editor.
  * @private
  */
 Blockly.FieldTextInput.prototype.widgetDispose_ = function() {
-  // Finalize value.
+  this.onEditorClose_();
+  this.unbindInputEvents_();
+  var style = Blockly.WidgetDiv.DIV.style;
+  style.width = 'auto';
+  style.height = 'auto';
+  style.fontSize = '';
+  this.htmlInput_ = null;
+};
+
+/**
+ * Deals with the not-disposal parts of closing the text input field's editor.
+ * @private
+ */
+Blockly.FieldTextInput.prototype.onEditorClose_ = function() {
```

Comment:
I don't see a reason to split this. 
There's already a divide here between widgetCreate and widgetDispose. 
I would however be in favour of adding a disposeView_ to match initView_ and dispose of any UI element references there.

Reply:
Just add comments in widgetDispose_ for a cleaner structure.